src/app/use-cases/validate-document.ts

```diff
@@ -67,13 +65,17 @@ export const makeValidateDocument =
         pipe(
           request,
           addUrlToDocument(payload.documentId, payload.documentUrl),
-          E.map((documents) => ({ ...request, documents }))
+          E.map((documents) => ({ ...request, documents })),
+          E.map((request) => ({
+            ...request,
+            status:
+              request.documents.every(
+                (document) => document.url && document.url.length > 0
+              ) && request.status === "DRAFT"
```

Comment:
In this case the change of status is also linked to the presence of the documents. A possible implementation could be:
`const status = (request: SignatureRequest) => Status`. 
Otherwise I didn't understand.

Reply:
I mean that all constraints on status update (from the previous state) must be isolated. What we want to avoid is invalid status changes.

If the changes depends on the whole signature request (and only on that) you may use 

`const nextStatus = (request: SignatureRequest) => Status`

otherwise `const next = (prev: Status) => Status` is enough.